### PR TITLE
Create requestIdleCallback preference for throttleScroll and ad-loading

### PR DESCRIPTION
**Pull request by sndrs**
Allows you to opt-in to a version of the site that delays `throttledScroll` and all commercial dom manipulation until an `idleCallback` is available (if supported).

This is to test the effect of `requestIdleCallback` on the FPS/interaction hit we have currently at load.

cc @uplne @Calanthe @jbreckmckye @paullewis

Opt-in would be available via http://www.theguardian.com/uk#gu.prefs.use-idle-callback=true

### CODE REVIEW

File: static/src/javascripts/bootstraps/common.js [hunk 1]
```diff
@@ -238,18 +238,19 @@ define([
             // However, this means it's VITAL that all writes in callbacks are delegated to fastdom
             throttledScrollEvent: function () {
                 var running = false;
-                window.addEventListener('scroll', function () {
+                var emit = function () {
                     if (!running) {
                         running = true;
-                        // Use fastdom to avoid forced synchronous layout:
```
Comment (jbreckmckye) [inline static/src/javascripts/bootstraps/common.js]: Isn't this comment still true if we're not using RIC? (Look, I already created a cool new acronym for it)

Comment (sndrs) [inline static/src/javascripts/bootstraps/common.js]: yeah it just seemed unnecessary

File: static/src/javascripts/projects/common/modules/commercial/article-body-adverts.js [hunk 2]
```diff
@@ -75,7 +75,7 @@ define([
 
                 ads.push($ad);
                 return new Promise(function (resolve) {
-                    fastdom.write(function () {
+                    idleFastdom.write(function () {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/article-body-adverts.js]: Just a heads up - I've actually been looking at this very write today. It seems the way we're finding spaces and writing elements asynchronously can create some race conditions. That means we might have to either do some reads and writes synchronously, or do some careful scheduling that prevents ads being injected into spaces that are no longer empty. 

If you're doing some performance investigation, though, I'm wary of doing anything that might add another variable to your testing.

**Comment by jbreckmckye** [comment]
Very exciting stuff.

:+1:

**Pull request merged**